scripts/githooks/commit_msg.py

```diff
@@ -0,0 +1,183 @@
+        return not self.title.endswith('.')
+
+    common_first_words = [ 'Add',
+                           'Adjust',
+                           'Support',
+                           'Change',
+                           'Remove',
+                           'Fix',
+                           'Print',
+                           'Track',
+                           'Refactor',
+                           'Combine',
+                           'Release',
+                           'Set',
+                           'Stop',
+                           'Make',
+                           'Enable',
+                           'Check',
+                           'Exclude']
+
+    def check_subject_imperative(self):
+        'Rule 5: Use the imperative mood in the subject line.'
+        'We can only check for common mistakes here, like using'
+        'the -ing form of a verb or non-imperative version of '
+        'common verbs'
+
+        firstword = self.title_words[0]
+
+        if firstword.endswith('ing'):
+            return False
+
+        for word in self.common_first_words:
+            if firstword.startswith(word) and firstword != word:
+                return False
+
+        return True
+
+    def check_body_limit(self):
+        'Rule 6: Wrap the body at 72 characters'
+
+        if len(self.body_lines) == 0:
+            return True
+
+        for line in self.body_lines:
+            if len(line) > 72:
+                return False
+
+        return True
+
+    def check_body_uses_why(self):
+        'Rule 7: Use the body to explain what and why vs. how'
+        # Not enforcable
+        return True
+
+    rule_funcs = [
+        check_subject_body_separtor,
+        check_subject_limit,
+        check_subject_capitalized,
+        check_subject_no_period,
+        check_subject_imperative,
+        check_body_limit,
+        check_body_uses_why ]
+
+    def check_the_seven_rules(self):
+        'validates the commit message against the seven rules'
+
+        num_violations = 0
+
+        for i in range(len(self.rule_funcs)):
+            func = self.rule_funcs[i]
+            res = func(self)
+            self.valid_rules[i] = res
+
+            if not res:
+                num_violations += 1
+
+        if num_violations > 0:
+            print
+            print "**** WARNING ****"
+            print
+            print "The commit message does not seem to comply with the project's guidelines."
+            print "Please try to follow the \"Seven rules of a great commit message\":"
+            print "https://chris.beams.io/posts/git-commit/"
+            print
+            print "The following rules are violated:\n"
+
+            for i in range(len(self.rule_funcs)):
+                if not self.valid_rules[i]:
+                    print "\t* Rule %d: \"%s\"" % (i+1, self.rules[i])
+
+        # Extra sanity checks beyond the seven rules
+        if len(self.body_lines) == 0:
```

Comment:
For these three checks can we exclude anything that has "Format" "Spelling" or "Documentation" in the title. These are common things that don't need explanations and where the title can be short.

Reply:
These are only warnings/notes, so harmless. I think having single or two word subjects makes it  highly likely that it would be possible to produce a better subject line.